src/overlays/actors/ovl_Bg_Hidan_Firewall/z_bg_hidan_firewall.c

```diff
@@ -21,23 +22,239 @@ const ActorInit Bg_Hidan_Firewall_InitVars =
+    0x00, 0x00, 0x00, 0x00,
+    0x19, 0x00, 0x01, 0x00,
+    0x001E,
+    0x0053,
+    0x00000000000000000000,
+};
+
+static SubActor98Init actor98InitData =
+{
+    0x01, 0x0050, 0x0064, 0xFF,
+};
+
+u32 D_80886D04[] = {
+    0x06015D20, 
+    0x06016120, 
+    0x06016520, 
+    0x06016920, 
+    0x06016D20, 
+    0x06017120, 
+    0x06017520, 
+    0x06017920, 
+    0x00000000, 
+    0x00000000, 
+    0x00000000,
+};
+
+void BgHidanFirewall_Init(BgHidanFirewall* this, GlobalContext* globalCtx){
+  
+    f32 scale;
+    
+    scale = 0.119999997318f;
+    this->actor.scale.x = scale;
+    this->actor.scale.z = scale;
+
+    scale = 0.00999999977648f;
+    this->actor.scale.y = 0.00999999977648f;
+
+    this->unk_150 = 0;
+
+    ActorCollider_AllocCylinder(globalCtx, &this->collider);
+    ActorCollider_InitCylinder(globalCtx, &this->collider, &this->actor, &cylinderInitData);
+
+    this->collider.dim.position.y = this->actor.posRot.pos.y;
+
+    func_80061ED4(&this->actor.sub_98, NULL, &actor98InitData);
+    
+    this->actionFunc = BgHidanFirewall_Wait;
+    return;
+}
+
+void BgHidanFirewall_Destroy(BgHidanFirewall* this, GlobalContext* globalCtx)
+{
+    BgHidanFirewall* thing = this;
+    ActorCollider_FreeCylinder(globalCtx, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Firewall/func_80886728.s")
+s32 BgHidanFirewall_Proximity(BgHidanFirewall* this, GlobalContext* globalCtx)
+{   
+    Player* player;
+    Vec3f distance;
+    
+    player = PLAYER;
+    func_8002DBD0(&this->actor, &distance, &player->actor.posRot.pos);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Firewall/func_80886794.s")
+    if (fabsf(distance.x) < 100.0f)
+    {
+        if (fabsf(distance.z) < 120.0f)
+        {   
+            return 1;
+        }
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Firewall/func_808867DC.s")
+void BgHidanFirewall_Wait(BgHidanFirewall* this, GlobalContext* globalCtx)
+{
+    if (BgHidanFirewall_Proximity(this, globalCtx) != 0)
+    {
+        this->actor.draw = BgHidanFirewall_Draw;
+        this->actor.params = 5;
+        this->actionFunc = BgHidanFirewall_Countdown;
+    }
+}
+
+void BgHidanFirewall_Countdown(BgHidanFirewall* this, GlobalContext* globalCtx)
+{
+
+    if (this->actor.params != 0)
+    {
+        this->actor.params--;
+    }
+    if (this->actor.params == 0)
+    {
+        this->actionFunc = BgHidanFirewall_Erupt;
+    }
+}
+
+void BgHidanFirewall_Erupt(BgHidanFirewall* this, GlobalContext* globalCtx)
+{
+    if (BgHidanFirewall_Proximity(this, globalCtx) != 0)
+    {
+        Math_ApproxF(&this->actor.scale.y, 0.1f, 0.024999999f);
+        return;
+    }
+    if (Math_ApproxF(&this->actor.scale.y, 0.00999999977648f, 0.00999999977648f) != 0)    
+    {
+        this->actor.draw = NULL;
+        this->actionFunc = BgHidanFirewall_Wait;
+    }
+    else
+    {
+        this->actor.params = 0;
+    }
+}
```

Comment:
```suggestion
void BgHidanFirewall_Erupt(BgHidanFirewall* this, GlobalContext* globalCtx)
{
    if (BgHidanFirewall_Proximity(this, globalCtx) != 0)
    {
        Math_ApproxF(&this->actor.scale.y, 0.1f, 0.024999999f);
    }
    else
    {
        if (Math_ApproxF(&this->actor.scale.y, 0.00999999977648f, 0.00999999977648f) != 0)    
        {
            this->actor.draw = NULL;
            this->actionFunc = BgHidanFirewall_Wait;
        }
        else
        {
            this->actor.params = 0;
        }
    }
}
```
The early return there is probably better off as an else like so